File: nodejs/demo/data-master/2-jabatan.mjs
/**
 * DEMO: Daftar jabatan
 *
 * Jalankan lewat terminal:  node 2-jabatan.mjs
 *
 * ID jabatan dipakai saat menambah karyawan (hr_job_position_id).
 */

import { apiGet, mulaiDemo, tampilkan, tulis } from '../../gajihub.mjs';

mulaiDemo('Daftar Jabatan');

const hasil = await apiGet('/hr/jobPositions');

if (hasil.sukses) {
    for (const jabatan of hasil.data.data) {
        // parent_id = ID jabatan atasan (kosong jika jabatan paling atas)
        tulis(`ID ${jabatan.id} : ${jabatan.name} (atasan: ${jabatan.parent_id ?? '-'})`);
    }
    tulis('');
}

tampilkan(hasil);

File: python/demo/data-master/2-jabatan.py
"""
DEMO: Daftar jabatan

Jalankan lewat terminal:  python 2-jabatan.py

ID jabatan dipakai saat menambah karyawan (hr_job_position_id).
"""

import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[2]))  # agar gajihub.py bisa di-import

from gajihub import api_get, mulai_demo, tampilkan, tulis

mulai_demo('Daftar Jabatan')

hasil = api_get('/hr/jobPositions')

if hasil['sukses']:
    for jabatan in hasil['data']['data']:
        # parent_id = ID jabatan atasan (kosong jika jabatan paling atas)
        atasan = jabatan.get('parent_id')
        tulis(f"ID {jabatan['id']} : {jabatan['name']} (atasan: {'-' if atasan is None else atasan})")
    tulis('')

tampilkan(hasil)
